### PR TITLE
[nrf noup] Fix blocker issue with stack sizes

### PR DESCRIPTION
**Pull request by ArekBalysNordic**
The stack sizes are too low when Oberon and CC3XX drivers are enabled.

### CODE REVIEW

File: config/nrfconnect/chip-module/Kconfig.defaults [hunk 1]
```diff
@@ -229,6 +229,7 @@ config IEEE802154_NRF5_RX_STACK_SIZE
     default 1024
 
 config OPENTHREAD_THREAD_STACK_SIZE
+    default 6144 if PSA_CRYPTO_DRIVER_CC3XX && OBERON_BACKEND
```
Comment (LuDuda) [inline config/nrfconnect/chip-module/Kconfig.defaults:232]:
```suggestion
    default 6144 if PSA_CRYPTO_DRIVER_CC3XX && PSA_CRYPTO_DRIVER_OBERON
```

And same for other?

**Review by LuDuda — APPROVED** [review]

File: src/platform/nrfconnect/CHIPDevicePlatformConfig.h [hunk 2]
```diff
@@ -215,11 +215,7 @@
 #endif // CHIP_DEVICE_CONFIG_CHIP_TASK_PRIORITY
 
 #ifndef CHIP_DEVICE_CONFIG_CHIP_TASK_STACK_SIZE
-#ifdef CONFIG_LTO
```
Comment (Damian-Nordic) [inline src/platform/nrfconnect/CHIPDevicePlatformConfig.h:218]: Why this change? It's not mentioned in the PR description.

File: config/nrfconnect/chip-module/Kconfig.defaults [hunk 3]
```diff
@@ -229,6 +229,7 @@ config IEEE802154_NRF5_RX_STACK_SIZE
     default 1024
 
 config OPENTHREAD_THREAD_STACK_SIZE
+    default 6144 if PSA_CRYPTO_DRIVER_CC3XX && PSA_CRYPTO_DRIVER_OBERON
```
Comment (Damian-Nordic) [inline config/nrfconnect/chip-module/Kconfig.defaults:232]: I know we may be in hurry, but shouldn't OpenThread provide the valid defaults for this option?

**Review by Damian-Nordic — APPROVED** [review]

File: config/nrfconnect/chip-module/Kconfig.defaults (hunk 3 above)
Comment (ArekBalysNordic) [reply config/nrfconnect/chip-module/Kconfig.defaults:232]: I don't know if they have such a case when they use both drivers at the same time. We fell into this issue when we enabled support for the `ECDSA_SECP_K1_256` curve.

Comment (ArekBalysNordic) [reply config/nrfconnect/chip-module/Kconfig.defaults:232]: So it is a very specific case.

File: src/platform/nrfconnect/CHIPDevicePlatformConfig.h (hunk 2 above)
Comment (ArekBalysNordic) [reply src/platform/nrfconnect/CHIPDevicePlatformConfig.h:218]: We need to increase the stack size to 8k anyway, so it doesn't matter whether it is the LTO build or not.

Comment (ArekBalysNordic) [reply src/platform/nrfconnect/CHIPDevicePlatformConfig.h:218]: But... It is regarding CC + Oberon only, so I will add the ifdef there to not impact on the memory requirements table,

**Comment by ArekBalysNordic** [comment]
The unit test fails are not related to this PR.

**Review by markaj-nordic — APPROVED** [review]

**Review by Damian-Nordic — APPROVED** [review]
